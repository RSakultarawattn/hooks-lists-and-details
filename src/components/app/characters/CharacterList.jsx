import React from 'react';
import propTypes from 'prop-types';
import Character from './Character';
import { Link } from 'react-router-dom';


const CharacterList = ({ characters }) => {
  const characterElements = characters.map(character => ( 
    <>
      <Link to={`/DetailPage/${character.id}`}>
        <li key={character.id}>
          <Character {...character} />
        </li>
      </Link>
    </>
  ));
  return (
    <ul data-testid="characters">
      {characterElements}
    </ul>
  );
};
CharacterList.propTypes = {
  characters: propTypes.arrayOf(propTypes.shape({
    id: propTypes.number.isRequired,
    name: propTypes.string.isRequired,
    image: propTypes.string.isRequired
  })).isRequired
};
export default CharacterList;
