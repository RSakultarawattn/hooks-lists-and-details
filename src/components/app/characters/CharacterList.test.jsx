import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CharacterList from './CharacterList';

describe('Character item component', () => {
  afterEach(() => cleanup());
  it('renders a character', () => {
    const { asFragment } = render(<CharacterList
      characters={[]} 
    />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});