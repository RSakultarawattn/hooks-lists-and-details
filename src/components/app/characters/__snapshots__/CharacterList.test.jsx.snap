// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character item component renders a character 1`] = `
<DocumentFragment>
  <ul
    data-testid="characters"
  />
</DocumentFragment>
`;
